astropy/stats/jackknife.py

```diff
@@ -1,12 +1,21 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 import numpy as np
 
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from numpy.typing import NDArray
+
 __all__ = ["jackknife_resampling", "jackknife_stats"]
 __doctest_requires__ = {"jackknife_stats": ["scipy"]}
 
 
-def jackknife_resampling(data):
+def jackknife_resampling(data: NDArray) -> NDArray:
```

Comment:
Dtype hint. Even Any is fine.

Reply:
Fixed in [d94e763](https://github.com/astropy/astropy/pull/16562/commits/d94e76304f84c996d54ed53ebc75acdd8ae6327d)